samples/demo-server/src/customs.rs

```diff
@@ -0,0 +1,199 @@
+use std::sync::Arc;
+
+use opcua::{
+    nodes::{DataTypeBuilder, ObjectBuilder, ReferenceDirection, VariableBuilder},
+    server::node_manager::memory::SimpleNodeManager,
+    types::{
+        DataTypeDefinition, DataTypeId, EnumDefinition, EnumField, NodeId, ObjectId,
+        ReferenceTypeId, StructureDefinition, StructureField, StructureType,
+    },
+};
+
+pub fn add_custom_types(nm: Arc<SimpleNodeManager>, ns: u16) {
+    let enum_id = add_enum_data_type(&nm, ns);
+    let struct_id = add_custom_data_type(&nm, ns, &enum_id);
+
+    let addr = nm.address_space();
+    let mut addr = addr.write();
+    let error_node = NodeId::next_numeric(ns);
+    let _error_data = ErrorData::new("No Error", 0, AxisState::Idle);
+    VariableBuilder::new(&error_node, "stErrorData", "stErrorData")
+        .organized_by(ObjectId::ObjectsFolder)
+        .data_type(struct_id.clone())
+        //.value(Variant::ExtensionObject::ST_ErrorData::default())
```

Comment:
@einarmo how do I do that? I am really unsure

Reply:
Just `.value(ExtensionObject::new(ST_ErrorData::default())` should work.

Reply:
that leads to the error:

   |
23  |         .value(ExtensionObject::new(error_data))
    |                -------------------- ^^^^^^^^^^ the trait `opcua::opcua_types::MessageInfo` is not implemented for `ErrorData`, which is required by `ErrorData: DynEncodable`
    |                |
    |                required by a bound introduced by this call


I can add implementation for these traits. But all other structs in opcua code seem to only  derive from `#[derive(Debug, Clone, PartialEq, opcua::types::BinaryEncodable, opcua::types::BinaryDecodable)]` as I do in that example. Am I missing something? any macro available?

Reply:
ok . they do not derive anything because they all implement that by hand... now I just tneed to remember what these values should be...

```
impl opcua::types::MessageInfo for WriteValue {
    fn type_id(&self) -> opcua::types::ObjectId {
        opcua::types::ObjectId::WriteValue_Encoding_DefaultBinary
    }
    fn json_type_id(&self) -> opcua::types::ObjectId {
        opcua::types::ObjectId::WriteValue_Encoding_DefaultJson
    }
    fn xml_type_id(&self) -> opcua::types::ObjectId {
        opcua::types::ObjectId::WriteValue_Encoding_DefaultXml
    }
    fn data_type_id(&self) -> opcua::types::DataTypeId {
        opcua::types::DataTypeId::WriteValue
    }
}
```

Reply:
Well I am really not sure how to implement that... the custom structs Ids are not in the ObjectId enum for sure since I create them.
@einarmo ...

Reply:
Implement `ExpandedMessageInfo` instead, then you can return `ExpandedNodeId`s pointing to arbitrary namespaces, you don't even need to provide a namespace index, you can just use the full namespace.

Reply:
The rust compiler is sometimes a little eager about inferring which traits you need to implement. `ExpandedMessageInfo` is implemented for all types that implement `MessageInfo`, but only core OPC-UA types can really implement `MessageInfo`.

Reply:
ok ExpandedNodeId is easier to return...
for the sample I can do whatever, 
However in my case the namespace it defined at runtime by reading a config. I am really unsure how I can implement that without very ugly tricks..

Reply:
I guess you could store it in a static? It's a bit of a niche requirement. Usually if the structure is static, meaning you define it at compile time, so is the namespace that defines the type.

Something like `static MY_NAMESPACE: OnceCell<String>` could work, if storing it in a static is acceptable. If the namespace _index_ is static you can return that instead. You could store it in the struct itself, but that will be hacky and potentially difficult to make work, so I wouldn't recommend it.

We need the namespace index in order to encode the extension object.

Reply:
I don't know what you're making or why, but I would recommend defining types and other shared components in a separate, static namespace the way all the companion standards do it.